File: tests/fixtures/semgrep/no-eval.py
# Tests for no-eval-dynamic-exec-python rule
import ast
import json

# ruleid: no-eval-dynamic-exec-python
result = eval(user_input)
# ruleid: no-eval-dynamic-exec-python
exec(code_string)

# ok: no-eval-dynamic-exec-python
data = json.loads(raw_json)
# ok: no-eval-dynamic-exec-python
value = ast.literal_eval(literal_string)

File: tests/fixtures/semgrep/no-eval.js
// Tests for no-eval-dynamic-exec rule

// ruleid: no-eval-dynamic-exec
const result = eval(userInput);
// ruleid: no-eval-dynamic-exec
const fn = new Function("return " + code);
// ruleid: no-eval-dynamic-exec
const fn2 = Function("alert(1)");

// ok: no-eval-dynamic-exec
const data = JSON.parse(rawJson);
// ok: no-eval-dynamic-exec
const config = { eval: false };
// ok: no-eval-dynamic-exec
const name = "eval";
